couchbase_test.go

```diff
@@ -0,0 +1,694 @@
+
+	if err := testCredsExist(t, username, password); err != nil {
+		t.Fatalf("Could not connect with new credentials: %s", err)
+	}
+
+	err = testRevokeUser(t, username)
+	if err != nil {
+		t.Fatalf("Could not revoke user: %s", username)
+	}
+}
+func TestCouchbaseDB_RotateRootCredentials(t *testing.T) {
+	if os.Getenv("VAULT_ACC") == "" {
+		t.SkipNow()
+	}
+	log.Printf("Testing RotateRootCredentials()")
+	_, address, port := prepareCouchbaseTestContainer(t)
+
+	connectionDetails := map[string]interface{}{
+		"hosts":    address,
+		"port":     port,
+		"username": "rotate-root",
+		"password": "rotate-rootpassword",
+	}
+	if pre_6dot5 {
+		connectionDetails["bucket_name"] = "foo"
+	}
+
+	db := new()
+	_, err := db.Init(context.Background(), connectionDetails, true)
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	if !db.Initialized {
+		t.Fatal("Database should be initialized")
+	}
+
+	statements := []string{""}
+
+	password, err := db.RotateRootCredentials(context.Background(), statements)
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	if err := testCredsExist(t, db.Username, password["password"].(string)); err != nil {
+		t.Fatalf("Could not connect with new RotatedRootcredentials: %s", err)
+	}
+	// Set password back
+	testCouchbaseDB_SetCredentials(t, "rotate-root", "rotate-rootpassword")
+
+	db.Close()
+
+}
+
+func testCouchbaseDB_SetCredentials(t *testing.T, username, password string) {
+
+	_, address, port := prepareCouchbaseTestContainer(t)
+
+	log.Printf("Testing SetCredentials()")
+
+	connectionDetails := map[string]interface{}{
+		"hosts":    address,
+		"port":     port,
+		"username": "Administrator",
+		"password": "Admin123",
+	}
+	if pre_6dot5 {
+		connectionDetails["bucket_name"] = "foo"
+	}
+
+	db := new()
+	_, err := db.Init(context.Background(), connectionDetails, true)
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	if !db.Initialized {
+		t.Fatal("Database should be initialized")
+	}
+
+	statements := dbplugin.Statements{}
+
+	// test that SetCredentials fails if the user does not exist...
+
+	staticUser := dbplugin.StaticUserConfig{
+		Username: "userThatDoesNotExist",
+		Password: password,
+	}
+
+	_, _, err = db.SetCredentials(context.Background(), statements, staticUser)
+	if err == nil {
+		t.Fatalf("err: did not error on setting password for userThatDoesNotExist")
+	}
+
+	staticUser = dbplugin.StaticUserConfig{
+		Username: username,
+		Password: password,
+	}
+
+	username, password, err = db.SetCredentials(context.Background(), statements, staticUser)
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	db.Close()
+
+	if err := testCredsExist(t, username, password); err != nil {
+		t.Fatalf("Could not connect with rotated credentials: %s", err)
+	}
+}
+
+func TestCouchbaseDB_SetCredentials(t *testing.T) {
+	if os.Getenv("VAULT_ACC") == "" {
+		t.SkipNow()
+	}
+
+	testCouchbaseDB_SetCredentials(t, "vault-edu", "password")
+}
+
+/*
+func TestCouchbaseDB_StaticRole_Rotations(t *testing.T) {
```

Comment:
I forgot one comment, looks like this was reference material while you were writing tests; safe to delete now?

Reply:
Deleted.